yt/data_objects/static_output.py

```diff
@@ -1984,6 +1984,18 @@ def __init__(
             default_species_fields=default_species_fields,
         )
 
+    def _disable_periodicity_for_bbox(self):
+        # This is called when a bounding box is explicitly set by the
+        # user--currently in these circumstances periodicity behaves
+        # very oddly (see https://github.com/yt-project/yt/issues/2639),
+        # so to prevent buggy behavior we simply disable it.
+        if self._periodicity != (False,) * 3:
+            mylog.warning(
+                "A bounding box was explicitly specified, so we "
+                "are disabling periodicity."
+            )
+            self._periodicity = (False,) * 3
```

Comment:
actually, this logic could just be encapsulated inside the `periodicity` property, couldn't it ?
Then you wouldn't need to update anything else, as the change would already affect all frontends !

Reply:
You don't want to change all frontends--only the ones that accept a `bounding_box` argument

Reply:
Aren't those the same that use the optional attribute `_domain_override` ?

Reply:
Yes, but I'm not sure what you are getting at here.

Reply:
What I mean is that it seems to me that all of this could be accomplished with the `Dataset.periodicity` property definition using a `if getattr(self, "_domain_override", False):` clause. Is this incorrect ?

Reply:
Ah, ok--yes, that's correct.

Reply:
Here's exactly what I have in mind
```patch
diff --git a/yt/data_objects/static_output.py b/yt/data_objects/static_output.py
index 96d013992..328f66262 100644
--- a/yt/data_objects/static_output.py
+++ b/yt/data_objects/static_output.py
@@ -322,6 +322,14 @@ class Dataset(abc.ABC):
     def periodicity(self):
         if self._force_periodicity:
             return (True, True, True)
+        elif hasattr(self, "_domain_override"):
+            # dataset loaded with a bounding box
+            if any(self._periodicity):
+                mylog.warning(
+                    "A bounding box was explicitly specified, so we "
+                    "are disabling periodicity."
+                )
+            return (False, False, False)
         return self._periodicity

     def force_periodicity(self, val=True):
```

Reply:
Ok I've added it